eth/consensus/clique/clique.py

```diff
@@ -0,0 +1,173 @@
+import logging
+import time
+from typing import Sequence, Iterable
+
+from eth.abc import (
+    AtomicDatabaseAPI,
+    BlockHeaderAPI,
+    VirtualMachineAPI,
+)
+from eth.db.chain import ChainDB
+
+from eth_typing import (
+    Address,
+    Hash32,
+)
+from eth_utils import (
+    encode_hex,
+    to_tuple,
+    ValidationError,
+)
+
+from eth.rlp.headers import BlockHeader
+
+from eth.typing import (
+    HeaderParams,
+    VMConfiguration,
+    VMFork,
+)
+from eth.vm.chain_context import ChainContext
+from eth.vm.execution_context import (
+    ExecutionContext,
+)
+
+from .datatypes import (
+    Snapshot,
+)
+from .header_cache import HeaderCache
+from .snapshot_manager import SnapshotManager
+from ._utils import (
+    get_block_signer,
+    is_in_turn,
+    validate_header_integrity,
+)
+
+# Clique Consensus implementation according to
+# https://github.com/ethereum/EIPs/issues/225
+
+
+def create_execution_context(self: BlockHeaderAPI,
+                             prev_hashes: Iterable[Hash32],
+                             chain_context: ChainContext,
+                             import_block: BlockHeaderAPI=None) -> ExecutionContext:
+
+    # In Clique consensus, the tx fee goes to the signer
+    fee_recipient = None
+    if import_block is not None:
+        try:
+            fee_recipient = get_block_signer(import_block.header)
+        except Exception:
+            pass
+
+    return ExecutionContext(
+        coinbase=self.coinbase,
+        timestamp=self.timestamp,
+        block_number=self.block_number,
+        difficulty=self.difficulty,
+        gas_limit=self.gas_limit,
+        prev_hashes=prev_hashes,
+        chain_id=chain_context.chain_id,
+        fee_recipient=fee_recipient
+    )
+
+
+BlockHeader.create_execution_context = create_execution_context  # type: ignore
```

Comment:
eeek!  just leaving comment here because this will be problematic if left in.

Reply:
aYep, I know...but what's the alternative? Maybe `create_execution_context` should be moved (and changed to take in a header) to the VM and then this can be overwritten as part of the `vm.configure` step?

Reply:
Probably yes.